models/user/user.go

```diff
@@ -908,10 +908,19 @@ func UpdateUser(ctx context.Context, u *User, changePrimaryEmail bool, cols ...s
 			if _, err := e.Insert(&emailAddress); err != nil {
 				return err
 			}
-		} else if _, err := e.ID(emailAddress.ID).Cols("is_primary").Update(&EmailAddress{
-			IsPrimary: true,
-		}); err != nil {
-			return err
+		} else {
+			if emailAddress.UID != u.ID {
+				return ErrEmailAlreadyUsed{
+					Email: u.Email,
+				}
+			}
+
+			_, err := e.ID(emailAddress.ID).Cols("is_primary").Update(&EmailAddress{
+				IsPrimary: true,
+			})
+			if err != nil {
+				return err
+			}
```

Comment:
Just thought about a question, does it need to update the old primary emails to is_primary=false as line 898?

Reply:
Yes, good remark, just updated pr